h2o-core/src/main/java/water/parser/ParseDataset.java

```diff
@@ -715,7 +716,41 @@ private FVecParseWriter makeDout(ParseSetup localSetup, int chunkOff, int nchunk
       byte[] zips = vec.getFirstBytes();
       ZipUtil.Compression cpr = ZipUtil.guessCompressionMethod(zips);
       if (localSetup._check_header == ParseSetup.HAS_HEADER) { //check for header on local file
-        byte[] bits = decryptionTool.decryptFirstBytes(ZipUtil.unzipBytes(zips, cpr, localSetup._chunk_size));
+
+        int previewSize = _parseSetup._chunk_size;
+        int estimatedHeaderSize = 0;
+        if (_parseSetup._column_names != null) {
```

Comment:
We could do the size estimation only if header checking fails for the first time too and not compare it right away.